python/orchestrate/_internal/http_handler.py

```diff
@@ -25,6 +32,90 @@ def _get_additional_headers() -> Mapping[str, str]:
     return {}
 
 
+def _coalesce_nullable_value(element: Element, tag: str) -> str:
+    value = element.find(tag)
+    return value.attrib["value"] if value is not None else ""
+
+
+@dataclass
+class _OperationalOutcomeIssue:
+    severity: str
+    code: str
+    diagnostics: str
+
+    @staticmethod
+    def from_xml_element(element: Element) -> "_OperationalOutcomeIssue":
+        severity = _coalesce_nullable_value(element, "{http://hl7.org/fhir}severity")
+        code = _coalesce_nullable_value(element, "{http://hl7.org/fhir}code")
+        diagnostics = _coalesce_nullable_value(
+            element, "{http://hl7.org/fhir}diagnostics"
+        )
+        return _OperationalOutcomeIssue(severity, code, diagnostics)
+
+    def __str__(self) -> str:
+        return f"{self.severity}: {self.code} - {self.diagnostics}"
+
+
+def _read_xml_outcomes(response: requests.Response) -> list[_OperationalOutcomeIssue]:
+    try:
+        operation_outcome = ElementTree.fromstring(response.text)
+        return [
+            _OperationalOutcomeIssue.from_xml_element(issue)
+            for issue in operation_outcome.findall(".//{http://hl7.org/fhir}issue")
+        ]
+    except Exception:
+        pass
+
+    return []
+
+
+def _read_json_outcomes(response: requests.Response) -> list[_OperationalOutcomeIssue]:
+    try:
+        json_response = response.json()
+        if "issue" in json_response:
+            return [
+                _OperationalOutcomeIssue(
+                    issue.get("severity", ""),
+                    issue.get("code", ""),
+                    issue.get("diagnostics", ""),
+                )
+                for issue in json_response["issue"]
+            ]
+        if (
+            json_response.get("type")
+            == "https://tools.ietf.org/html/rfc9110#section-15.5.1"
+        ):
+            return [
+                _OperationalOutcomeIssue(
+                    severity="error",
+                    code=json_response.get("title", ""),
+                    diagnostics=json_response.get("detail", ""),
+                )
+            ]
+    except Exception:
+        pass
+
+    return []
+
+
+def _read_operational_outcomes(response: requests.Response) -> list[str]:
+    outcomes = _read_xml_outcomes(response)
+    if outcomes:
+        return [str(outcome) for outcome in outcomes]
+    outcomes = _read_json_outcomes(response)
+    if outcomes:
+        return [str(outcome) for outcome in outcomes]
+
+    return []
+
+
+def _exception_from_response(response: requests.Response) -> OrchestrateHttpError:
+    operational_outcomes = _read_operational_outcomes(response)
+    if str(response.status_code).startswith("4"):
```

Comment:
I believe Orchestrate may return some 50x responses with useful operational outcomes to consider for return to the client.

Reply:
Oh, that's good to know. Do you have any examples of this you can DM me?